### ISSUE TITLE
cmd/compile: overflow in int -> string

### ISSUE THREAD

**Issue by dvyukov** [golang/go#11330]
Gc rejects to compile the following program:
```go
package a
var a = string(9223372036854775808)
```
saying:
```
overflow in int -> string
```

go/types compile it successfully.

Compilers must agree on whether it is a valid Go program or not.

on commit af817890aa7b628234075a6cb01a3a435fb8317d

**Comment by griesemer** [comment]
Given the spec as is, I believe this should compile w/o errors. The spec says:

http://tip.golang.org/ref/spec#Conversions
```
A constant value x can be converted to type T in any of these cases:
...
x is an integer constant and T is a string type. The same rule as for non-constant x applies in this case.
```

and:
http://tip.golang.org/ref/spec#Conversions_to_and_from_a_string_type
```
Converting a signed or unsigned integer value to a string type yields a string containing the UTF-8 representation of the integer. Values outside the range of valid Unicode code points are converted to "\uFFFD".
```

There's no mention of a compile-time (or even run-time) error. The integer should be converted into "\uFFFD".

**Comment by griesemer** [comment]
Closed by mistake.

**Comment by dvyukov** [comment]
Then this is a bug in gc, right?

**Comment by griesemer** [comment]
@dvyukov yes, I believe this is a gc compiler bug.

We could make the spec more restrictive, but if the valid Unicode ranges changes, the compiler would have to be updated (that said, it would probably be a range extension and thus a backward-compatible change).

**Comment by rsc** [comment]
Too late for Go 1.5. The only possible visible effect of making changes to fix hypothetical bugs like this one is introducing real ones.

**Comment by griesemer** [comment]
This was fixed via https://go-review.googlesource.com/#/c/21344/ (fix for #15039).
